src/sage/crypto/historical/enigma.py

```diff
@@ -0,0 +1,1148 @@
+            current position 9 (Initially 9), offset of 6, notch at 22 and
+            using permutation [1, 3, 5, 7, 9, 11, 2, 15, 17, 19, 23, 21, 25,
+            13, 24, 4, 8, 22, 6, 0, 10, 12, 20, 18, 16, 14], Rotor with current
+            position 12 (Initially 12), offset of 8, notch at 10 and using
+            permutation [4, 18, 14, 21, 15, 25, 9, 0, 24, 16, 20, 8, 17, 7, 23,
+            11, 13, 5, 19, 6, 10, 3, 2, 12, 22, 1], Rotor with current position
+            15 (Initially 15), offset of 10, notch at 26 and using permutation
+            [21, 25, 1, 17, 6, 8, 19, 24, 20, 15, 18, 3, 13, 7, 11, 23, 0, 22,
+            12, 9, 16, 14, 5, 4, 2, 10], Rotor with current position 18
+            (Initially 18), offset of 12, notch at 10 and using permutation
+            [4, 18, 14, 21, 15, 25, 9, 0, 24, 16, 20, 8, 17, 7, 23, 11, 13, 5,
+            19, 6, 10, 3, 2, 12, 22, 1], Rotor with current position 21
+            (Initially 21), offset of 14, notch at 22 and using permutation
+            [1, 3, 5, 7, 9, 11, 2, 15, 17, 19, 23, 21, 25, 13, 24, 4, 8, 22,
+            6, 0, 10, 12, 20, 18, 16, 14], Rotor with current position 24
+            (Initially 24), offset of 16, notch at 5 and using permutation
+            [0, 9, 3, 10, 18, 8, 17, 20, 23, 1, 11, 7, 22, 19, 12, 2, 16, 6,
+            25, 13, 15, 24, 5, 21, 14, 4], Rotor with current position 26
+            (Initially 26), offset of 18, notch at 17 and using permutation
+            [4, 10, 12, 5, 11, 6, 3, 16, 21, 25, 13, 19, 14, 22, 24, 7, 23,
+            20, 18, 15, 0, 8, 1, 17, 2, 9]], Reflector or Plugboard using
+            permutation [5, 21, 15, 9, 8, 0, 14, 24, 4, 3, 17, 25, 23, 22, 6,
+            2, 19, 10, 20, 16, 18, 1, 13, 12, 7, 11] as a reflector and None
+            as a plugboard
+
+        The length of the list for the initials position and the list for the
+        offsets must be the same length as the list for the rotors to be used
+        since every rotor need a initial position and offset::
+
+            sage: e = Enigma.create([5, 3, 2, 4], [5, 24], [2, 4, 24], 2, None)
+            Traceback (most recent call last):
+            ...
+            ValueError: initial_positions and rotors must be the same length
+            sage: e = Enigma.create([5, 3, 2], [5, 24, 1], [2, 4], 2, None)
+            Traceback (most recent call last):
+            ...
+            ValueError: offsets and rotors must be the same length
+
+        There are not unlimited predefined rotors and reflectors::
+
+            sage: e = Enigma.create([2, 6], [2, 5], [12, 13], 2, None)
+            Traceback (most recent call last):
+            ...
+            ValueError: Values in rotors must be a integer between 1 and 5
+            sage: e = Enigma.create([2, 5], [2, 5], [12, 13], 4, None)
+            Traceback (most recent call last):
+            ...
+            ValueError: Reflector must be a value between 1 and 3
+        """
+
+        wheels = [
```

Comment:
Does it fit on one line? Not a big deal though. We use a maximum line length of 80 characters.